vpr/src/base/clustered_netlist.cpp

```diff
@@ -3,13 +3,15 @@
 #include "vtr_assert.h"
 #include "vpr_error.h"
 
+#include <utility>
+
 /**
  * @file
  * @brief ClusteredNetlist Class Implementation
  */
 
 ClusteredNetlist::ClusteredNetlist(std::string name, std::string id)
```

Comment:
I don't understand how it can be dangerous. If the user passes an rvalue string, we take the ownership of its copy as the argument is passed by value, and the string used by the caller will persist in its scope. In case of an lvalue string, the object would be destroyed immediately if we don't take its ownership. I think using std::move in this context is similar to implementing ClusteredNetlist(const std::string& name, const std::string& id) and ClusteredNetlist(std::string&& name, std::string&& id) and two other variatios of the constructor separatly.

Reply:
[This](https://stackoverflow.com/questions/51705967/advantages-of-pass-by-value-and-stdmove-over-pass-by-reference) explains why passing strings by value and using std::move is recommended.

Reply:
Ah, I missed that you were passing the strings into the constructor by value; I assumed they were being passed by reference and then you used std::move to grab the contents.